### PR TITLE
FSE: Properly namespace all PHP

### LINKED ISSUE #34795: FSE: prefix function names and follow plugin best practices
Let's remember to prefix function names, `a8c_` should probably do the trick, but let us know if folks had other suggestions.

https://developer.wordpress.org/plugins/plugin-basics/best-practices/#prefix-everything

cc @Automattic/ajax @Automattic/cylon @Automattic/serenity

### PR DESCRIPTION
**Pull request by obenland**
Adds a common namespace.
See https://developer.wordpress.org/plugins/plugin-basics/best-practices/#prefix-everything

This is largely untested. So far I've only made sure that the plugin doesn't fatal and SPT works as expected. The FSE side definitely needs more testing.

Fixes #34795

### CODE REVIEW

**Comment by gwwar** [comment]
@obenland if you rebase, I'll try and quickly help smoke test.

**Comment by obenland** [comment]
> @obenland if you rebase, I'll try and quickly help smoke test.

It looks like _this_ rebase went through as expected.

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php [hunk 1]
```diff
@@ -8,22 +8,24 @@
  * License: GPLv2 or later
  * Text Domain: full-site-editing
  *
- * @package full-site-editing
+ * @package A8C\FSE
  */
 
+namespace A8C\FSE;
```
Comment (gwwar) [inline apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php]: Namespaces! 🚀

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/utils/class-wp-template.php [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
  */
 
+namespace A8C\FSE;
+
 /**
- * Class A8C_WP_Template
+ * Class WP_Template
  */
-class A8C_WP_Template {
+class WP_Template {
```
Comment (vindl) [inline apps/full-site-editing/full-site-editing-plugin/full-site-editing/utils/class-wp-template.php]: Just noting that we'll likely need to update the usage here https://github.com/Automattic/themes/blob/master/modern-business/header.php#L28 and similar one in `footer.php`.

Comment (gwwar) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/utils/class-wp-template.php]: @vindl do you prefer keeping this as A8C_WP_Template?

Comment (vindl) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/utils/class-wp-template.php]: No, I prefer the namespaced one. Updated in https://github.com/Automattic/themes/pull/1174/commits/811a2cab9b5665597ec80d3874fa0a2d78414ca8.

**Pull request merged**